### DISCUSSION TITLE
Occasional burst of HTTP 403 Errors when querying the GraphQL endpoint w/ app token

### DISCUSSION THREAD

**Discussion by jamesdh** [community/community#24494] [Apps, API and Webhooks]
<p>I’m working on a GitHub app that makes a fair number of calls to the GraphQL endpoint, and occasionally I’ll see the endpoint return HTTP 403 Forbidden, usually for several seconds at a time, occasionally for 20-30 seconds. There is no more detailed message returned.</p>
<p>Are there any common causes for this that are not documented? I’m not hitting my rate limit, and this is separate from the 5xx errors that one might get if the server takes too long to query GitHub’s own backend and times out.</p>
<p>My assumption is that this is some sort of undocumented restriction when querying the API too frequently in too short of a period of time. Is there any information regarding this anywhere?</p>

**Comment by francisfuzz** [comment, marked as answer]
<aside class="quote no-group" data-username="jamesdh" data-post="1" data-topic="125422">
<div class="title">
<div class="quote-controls"></div>
 jamesdh:</div>
<blockquote>
<p>Are there any common causes for this that are not documented? I’m not hitting my rate limit, and this is separate from the 5xx errors that one might get if the server takes too long to query GitHub’s own backend and times out.</p>
<p>My assumption is that this is some sort of undocumented restriction when querying the API too frequently in too short of a period of time. Is there any information regarding this anywhere?</p>
</blockquote>
</aside>
<p>:wave: <a class="mention" href="/jamesdh">@jamesdh</a>: It sounds like you’re not hitting the primary rate limit, but the secondary rate limit (also known as the anti-abuse rate limit).</p>
<p>The GraphQL API has anti-abuse rate limits just like its older sibling, the REST API (more on the <a href="https://docs.github.com/en/rest/overview/resources-in-the-rest-api#abuse-rate-limits">REST API’s abuse rate limits here</a>). Our guide covering best practices for integrators includes a section for <a href="https://docs.github.com/en/rest/guides/best-practices-for-integrators#dealing-with-abuse-rate-limits">how to deal with and avoid these abuse rate limits</a>. While we don’t have a more specific guide for the GraphQL API, our team is tracking an internal issue around adding documentation for these “secondary” limits.</p>
<p>We don’t give an exact number or a specific range for how many requests can (or should) be made to avoid the abuse-rate limits as these limits are subject to change at any time. If you’re making concurrent calls to the GitHub API, I suggest making fewer concurrent calls and either reduce or space them out if you’re still hitting the limit.</p>
<p>I’m wondering if that helps and if you have any more questions?</p>

**Reply by rocodcli** [reply]
👋  @francisfuzz 
when using the GraphQL API, 
how do you differentiate a 403 from the secondary rate limit and an actual rate limit due to lack of authorization?

**Comment by jamesdh** [comment]
<p>Thank you for the clarification <a class="mention" href="/francisfuzz">@francisfuzz</a>. That’s what I suspected, but still very helpful to have confirmation!</p>

**Comment by justkahdri** [comment]
Hey I'm having the same problem today, is there any documentation about it now about the graphQL API specifically?
